src/cli-client/internals/cli/cli_upload.go

```diff
@@ -0,0 +1,108 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/canonical/oci-factory/cli-client/internals/client"
+	"github.com/canonical/oci-factory/cli-client/internals/logger"
+	"github.com/canonical/oci-factory/cli-client/internals/trigger"
+)
+
+type UploadRelease struct {
+	Track     string
+	Risks     []string
+	EndOfLife string
+}
+
+type CmdUpload struct {
+	UploadRelease []string `long:"release" description:"Release images to container registries.\nSyntax: --release tracks=<release track>,risks=<risk1>[,<risk2>...],eol=yyyy-mm-dd"`
+}
+
+var cmdUpload CmdUpload
+var releases = make([]UploadRelease, 0)
+
+func init() {
+	// fmt.Println("DEBUG Adding upload to parser commands")
+	parser.Command.AddCommand("upload", "Trigger the build and release for the image in the current working directory",
+		"long description", &cmdUpload)
+}
+
+func (c *CmdUpload) Execute(args []string) error {
+	// fmt.Println("DEBUG CMD upload args:", args)
+	parseUploadReleases(c.UploadRelease)
+	triggerUploadReleases()
+	return nil
+}
+
+func parseUploadReleases(args []string) error {
+	var release UploadRelease
+	for _, argStr := range args {
+		// fmt.Println("DEBUG Upload release arg:", argStr)
+		parts := strings.Split(argStr, ",")
+
+		var risks []string
+		for _, part := range parts {
+			keyValue := strings.SplitN(part, "=", 2)
+			if len(keyValue) == 1 && len(risks) > 0 {
+				// This is part of the risks list (e.g., "stable" in "risks=beta,stable")
+				risks = append(risks, keyValue[0])
+				continue
+			}
```

Comment:
I dont have enough context about the parser but this looks like it is trying to catch the case where `risks=foo,bar` is at the end of the string. I have two suggestions here but I am not sure about them because I don't know the full details. I would recommend:
1. Having a comment describing the format.
2. Thinking of a way of making the parser a little bit more uniform. It does not have to be more complex necessarily but the format is a little bit ambiguous (commas and equals) so probably you need something a little bit more sophisticated. That can be regular expression or even a different way of splitting the string ( cannot think of good way now without regular expression though).

Reply:
Good suggestion with the format comment. 

For point 2, since we don't force the user input to be of a fixed order of `tracks`, `risks` and `eol`, a regular expression might not be a good idea. Therefore I think using an FSM-styled parser would better suit the need here.

Reply:
I don't have a strong opinion about regular expressions, if there is something simpler then that is indeed much better. However, just for the sake of completeness, you could do it with a regular expression with something like:
```
(((tracks=[^,]+)|(risks=([,^,]+,)*[^,]+)|(eol=[^,]+)),){0,2}(((tracks=[^,]+)|(risks=([,^,]+,)*[^,]+)|(eol=[^,]+)))
```
and then do some validation to see that there is only one `risks`, one `tracks`, etc.

Note: you can simplify the regex a lot by referring to previous capture group using `(?1)` and things like that.

Reply:
Tried out a bit. Would you think 
```
tracks=([^,]+),risks=([^,]+(?:,[^,]+)*),eol=(\d{4}-\d{2}-\d{2})
```
is a better idea?

Reply:
Would you think the following patter better?
```
(((tracks=[^,]+)|(risks=[^,]+,*[^,]+)|(eol=[^,]+)),?){3}
```

Reply:
The problem with that one is that it matches an input with a trailing comma at the end, right?

Reply:
But let's not use a regular expression if there is something simpler and the parsing code is easier to follow.

Reply:
Updated in [0c81222](https://github.com/canonical/oci-factory/pull/198/commits/0c81222bcd16b62d8d1ba393a0651fd3e5e584a9).